contracts/DomainUtilsAssembly.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+library DomainUtilsAssembly {
+    
+    function stripProtocol(string memory domain) internal pure returns (string memory) {
+        bytes memory httpsPrefix = "https://";
+        bytes memory httpPrefix = "http://";
+        
+        if (hasPrefix(domain, httpsPrefix)) {
+            return substring(domain, 8, bytes(domain).length);
+        } else if (hasPrefix(domain, httpPrefix)) {
+            return substring(domain, 7, bytes(domain).length);
+        }
+        return domain;
+    }
+
+    function extractParentDomain(string memory domain) internal pure returns (string memory) {
+        uint lastDot;
+        uint secondLastDot;
+        
+        lastDot = indexOf(domain, bytes1('.'), 0);
+        if (lastDot == type(uint256).max) return ""; 
+
+        secondLastDot = indexOf(domain, bytes1('.'), lastDot + 1);
+        if (secondLastDot == type(uint256).max) {
+            return substring(domain, lastDot + 1, bytes(domain).length);
+        }
+        return substring(domain, secondLastDot + 1, bytes(domain).length);
+    }
```

Comment:
It would be good to write tests for utils separately (via wrapper test contract, e.g.).
I can't understand how secondLastDot works does it mean that `test.news.com` returns just `com` ?

Reply:
fxd